src/xmpp/xmpp-im/types.cpp

```diff
@@ -2026,6 +2039,7 @@ bool Message::fromStanza(const Stanza &s, bool useTimeZoneOffset, int timeZoneOf
 
 	t = childElementsByTagNameNS(root, "http://jabber.org/protocol/muc#user", "x").item(0).toElement();
 	if(!t.isNull()) {
+		setMUCUser(true);
```

Comment:
What's wrong with writing directly to d->hasMUCUser? Having setMUCUser in public API does not make much sense anyway. I'd remove it.